File: kraken/src/main/java/glitch/kraken/services/BitsService.java
package glitch.kraken.services;

import glitch.api.http.Routes;
import glitch.api.objects.json.interfaces.OrdinalList;
import glitch.kraken.GlitchKraken;
import glitch.kraken.object.json.Cheermote;
import glitch.kraken.object.json.collections.Cheermotes;
import glitch.service.AbstractHttpService;
import reactor.core.publisher.Flux;

public class BitsService extends AbstractHttpService {
    public BitsService(GlitchKraken rest) {
        super(rest.getClient(), rest.getHttpClient());
    }

    public Flux<Cheermote> getCheermotes() {
        return exchangeTo(Routes.get("/bits/actions").newRequest(), Cheermotes.class)
                .flatMapIterable(OrdinalList::getData);
    }

    public Flux<Cheermote> getCheermotes(Long channelId) {
        return exchangeTo(Routes.get("/bits/actions").newRequest().queryParam("channel_id", channelId), Cheermotes.class)
                .flatMapIterable(OrdinalList::getData);
    }
}

File: kraken/src/main/kotlin/io/glitchlib/v5/service/BitsService.kt
package io.glitchlib.v5.service

import io.glitchlib.GlitchClient
import io.glitchlib.internal.http.bodyFlowable
import io.glitchlib.v5.internal.AbstractKrakenService
import io.glitchlib.v5.internal.model.json.Cheermotes
import io.glitchlib.v5.model.json.Cheermote
import io.reactivex.Flowable

class BitsService internal constructor(client: GlitchClient) : AbstractKrakenService(client) {
    fun getCheermotes(channelId: Long? = null): Flowable<Cheermote> =
            get<Cheermotes>("/bits/actions") {
                if (channelId != null) {
                    addQueryParameters("channel_id", channelId.toString())
                }
            }.bodyFlowable
}
